puppet/data/common.yaml

```diff
@@ -0,0 +1,33 @@
+foreman_servicename: 'foreman.theforeman.org'
+foreman_url: 'https://%{lookup("foreman_servicename")}'
+puppet_servicename: 'puppet.theforeman.org'
+
+foreman::repo::repo: '3.1'
+foreman::config::apache::proxy_no_proxy_uris:
+  - /icons
+  - /server-status
+foreman::serveraliases:
+  - '%{alias("foreman_servicename")}'
+
+foreman_proxy::trusted_hosts:
+  - "foreman01.conova.theforeman.org"
+foreman_proxy::foreman_base_url: '%{alias("foreman_url")}'
+
+puppet::runmode: cron
+puppet::puppetmaster: '%{alias("puppet_servicename")}'
+puppet::server_ca_allow_sans: true
+puppet::server_additional_settings:
+  dns_alt_names:
+    - '%{alias("puppet_servicename")}'
+puppet::server_foreman_url: '%{alias("foreman_url")}'
+puppet::server_environments_owner: 'deploypuppet'
+puppet::server_environments_group: 'deploypuppet'
+puppet::server_puppetserver_telemetry: false
+puppet::show_diff: true
+
+users::users:
+  ekohl:
+    fullname: 'Ewoud Kohl van Wijngaarden'
+
+# TODO:
+# - sysadmins
```

Comment:
@evgeni I think we should discuss how to set this up. Today it's done via Foreman's ENC. The problem is that you can't bootstrap with the ENC (unlike Hiera). Once you apply Puppet it prevents sudo access (since for each user an individual rule is made, unlike a group).

Including users and their password hashes in a public repo isn't exactly what you'd want. Also the sysadmins email addresses shouldn't be public. I was thinking about our setup and wonder if we can use the wheel group for sudo access.

For email access we could also enhance the user via `foreman_users` in the ENC. It'd mean we would need to set the owner correctly, but that's not a bad thing IMHO.

Reply:
> For email access we could also enhance the user via `foreman_users` in the ENC. It'd mean we would need to set the owner correctly, but that's not a bad thing IMHO.

https://github.com/theforeman/foreman-infra/pull/1789 would be that.

Reply:
`wheel` doesn't exist on Debian, but I'd rather go and implement it there (via Puppet) than have those individual files again ;)

I must admit I'm always a bit cautious with using ENC in "important" places (like mail or sudo), as that effectively means that if the ENC goes bonkers for some reason¹, Puppet will re-reploy the config with *no* mail aliases or *no* sudo access, which will make finding and fixing errors harder.

¹: I've seen this happening when rolling out the deploy-ssh-keys to jenkins nodes (or rather, their registration in web01), never with the sudo part, but still.

Reply:
Technically that was a Foreman search, not the ENC. Since 26ff6baf21a92fd6bbe3461f35f67a2d531f9bd0 via the ENC, though that did have a fallback.

Reply:
For now I've solved it by not including users and utility. This means we should continue setting that via Foreman's ENC for now.

What you would miss in a bootstrap then is the root email alias and the users. However, you have the initial user which is part of the wheel/sudo group and our new sudo rules respect that group. This means we can deal with it after the migration.